src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -945,6 +933,11 @@ private Task FillAsync(CancellationToken cancellationToken)
                 // The read completed synchronously, so update the amount of data in the buffer and return.
                 int bytesRead = t.GetAwaiter().GetResult();
                 if (NetEventSource.IsEnabled) Trace($"Received {bytesRead} bytes.");
+                if (bytesRead == 0)
+                {
+                    throw new IOException(SR.net_http_invalid_response);
```

Comment:
Consider adding a comment about why we throw in this case.  Presumably it's because FillAsync is only ever called when we're expecting more data, and so lack of any data means something went wrong.